### ISSUE TITLE
Revisit evaluation metrics and loss functions

### ISSUE THREAD

**Issue by isaacmg** [AIStream-Peelout/flow-forecast#19]
In the draft submitted to the NIPs workshop R^2 was chosen as the overall evaluation metric. However, increasingly it looks like R^2 may not be the [best choice](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4530125/). With that in mind it makes sense to evaluate alternatives for computing a score across multiple models of varying flow volumes.  Secondly, a better loss function help improve convergence and efficiency of transfer learning. The authors of the paper [Towards learning universal, regional, and local hydrological behaviors via machine learning applied to large-sample datasets](https://www.hydrol-earth-syst-sci.net/23/5089/2019/) came up with an NSE metric 
<img width="1105" alt="image" src="https://user-images.githubusercontent.com/3865062/71775420-c490cd00-2f4e-11ea-909a-9b7d72d63c0c.png">
 and loss function which apparently is commonly used by hydrologists . Also another recent paper studied [using shape and time distortion loss](https://papers.nips.cc/paper/8672-shape-and-time-distortion-loss-for-training-deep-time-series-forecasting-models).

 Other evaluation metrics could possibly include [normalized mean square error](https://rem.jrc.ec.europa.eu/RemWeb/atmes2/20b.htm), [MASE](https://rdrr.io/cran/Metrics/man/mase.html), or others.

 In summary for this issue the following should be accomplished: 

- [x] Determine advantages/disadvantages of each commonly used metric/loss with respect to comparing river flows that have ranges anywhere from 0-10cfs to 100,000-200,00cfs. Write brief literature summary in the Wiki.
- [ ] Add function to support computing of chosen metric(s) and loss functions. For info on writing [custom loss functions see here](https://discuss.pytorch.org/t/solved-what-is-the-correct-way-to-implement-custom-loss-function/3568)
- [ ] [Add custom loss functions and metrics to the dictionary](https://github.com/AIStream-Peelout/flow-forecast/blob/00f262556f37140dc9e7525e8dfa9b8624f67d0d/flood_forecast/model_dict_function.py#L12) to allow passing by parameter 
- [ ]  Add unit tests to insure code is correct (especially for any custom written computations that do not use a standard library). Each loss function should have its own unit test. Additionally an integration test should be added to ensure passing loss function/eval metric from the json config produces desired result. 

In the end both loss functions and evaluation metrics should be easily swappable in JSON config.

**Comment by isaacmg** [comment]
Adding a gaussian loss function
